cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -154,6 +154,42 @@ impl ABIMachineSpec for X64ABIMachineSpec {
                 );
             }
 
+            // Windows fastcall dictates that `__m128i` paramters to a function
+            // are passed indirectly as pointers, so handle that as a special
+            // case before the loop below.
+            if param.value_type.is_vector()
+                && param.value_type.bits() >= 128
+                && args_or_rets == ArgsOrRets::Args
+                && is_fastcall
+            {
+                let pointer = match get_intreg_for_arg(&call_conv, next_gpr, next_param_idx) {
+                    Some(reg) => {
+                        next_gpr += 1;
+                        ABIArgSlot::Reg {
+                            reg: reg.to_real_reg().unwrap(),
+                            ty: ir::types::I64,
+                            extension: ir::ArgumentExtension::None,
+                        }
+                    }
+
+                    // FIXME: this is implementable but is left for a future
+                    // commit. The pointer needs to become a stack argument, but
+                    // all the stack reservation for the actual values probably
+                    // needs to come before all the argument stack area, so
+                    // that refactoring is left for later.
```

Comment:
File a follow up issue to track this?

Reply:
Ah this ended up actually being necessary to fill out, so no longer relevant (sorry I probably pushed to the PR while you were reviewing)